### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by peppy**
I'm not 100% on the exception changes, but seems to be required.

The packages excluded from this bump all have issues that are high(er) effort or blocked.

### CODE REVIEW

**Comment by bdach** [comment]
Tested multiplayer and spectator full-stack, seems to work. Also tested exceptions (via sending a multi invite to a user who blocked sender), and also seems to work.

For whatever it's worth, https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051 claims that `[Serializable]` is also removable, but I'm not so sure about that so I'd rather not here.

Going to wait for CI to complete before merging but source seems ok.

**Review by bdach — APPROVED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**
